code/game/machinery/kitchen/microwave.dm

```diff
@@ -185,31 +189,36 @@
 VUEUI_MONITOR_VARS(/obj/machinery/microwave, microwavemonitor)
 	watch_var("operating", "on", CALLBACK(null, .proc/transform_to_boolean, FALSE))
 
-/obj/machinery/microwave/vueui_data_change(var/list/newdata, var/mob/user, var/datum/vueui/ui)
+/obj/machinery/microwave/vueui_data_change(var/list/data, var/mob/user, var/datum/vueui/ui)
 	var/monitordata = ..()
 	if(monitordata)
-		. = newdata = monitordata
+		. = data = monitordata
+
+	// init data objects if they aren't already
+	LAZYINITLIST(data["cookingobjs"])
+	LAZYINITLIST(data["cookingreas"])
 
-	newdata["cookingobjs"] = list()
-	newdata["cookingreas"] = list()
+	// set up the progress bar vars
+	VUEUI_SET_CHECK(data["cook_start"], start_time, ., data)
+	VUEUI_SET_CHECK(data["cook_elapsed"], world.time, ., data) // needed to update progress

```

Comment:
see I struggled with wtime for like thirty minutes and couldn't get it to work with the progress bar, and it turns out it's because I can't read and you need to do  `$root.$data.wtime` as you said below so honestly committing that there was a cry for help and I'm glad you came with deliverance